### PR TITLE
Refactored LanguageExtractor

### PR DESCRIPTION
**Pull request by lgfrbcsgo**
- Made `LanguageExtractor` more reusable by adding functionality to return the detected language

- Removed now unnecessary functionality in `LanguageExtractor` and `TestRuns`

- Worte some tests for `LanguageExtractor`

### CODE REVIEW

File: src/sample-data/sample-data-generator.ts [hunk 1]
```diff
@@ -54,7 +54,7 @@ export class SampleDataGenerator {
                     let p = new WebPage(data);
 
                     //Check if page is in english
-                    LanguageExtractor.isWebPageInLanguage(p, LanguageExtractor.ENGLISH_LANG_CODE, function (result: boolean) {
+                    LanguageExtractor.isWebPageInLanguage(p, "en", function (err, result) {
```
Comment (nyxathid) [inline src/sample-data/sample-data-generator.ts]: Does it work if I accidentally pass "eng" as a parameter? Something like that happened to me recently, that's why I added ENGLISH_LANG_CODE

File: src/language-extractor.ts [hunk 2]
```diff
@@ -1,126 +1,94 @@
...
 
-export class LanguageExtractor {
 
-    static path = require('path');
-    static fs = require('fs');
-    static cld = require('cld');
-    static WARCStream = require('warc');
+export class LanguageExtractor {
 
-    //On the test data only 0.6% pages are not classified, to detect these we can use franc ( execution time increases by only approximately 3% )
```
Comment (nyxathid) [inline src/language-extractor.ts]: that comment was quite informative ;)

File: src/test-runs.ts [hunk 3]
```diff
@@ -161,7 +94,7 @@ export class TestRuns {
 
             let tld = p.getTLD();
 
-            LanguageExtractor.isWebPageInLanguage(p, LanguageExtractor.ENGLISH_LANG_CODE, function(result : boolean) {
+            LanguageExtractor.isWebPageInLanguage(p, "en", function(err, result) {
```
Comment (nyxathid) [inline src/test-runs.ts]: Same as above: "eng" as a parameter?

Comment (lgfrbcsgo) [inline src/test-runs.ts]: Will throw an Error. The parameter is standardized under ISO 639-1. Basically 2 character language codes.

File: src/language-extractor.ts (hunk 2 above)
Comment (lgfrbcsgo) [inline src/language-extractor.ts]: I know, but I don't think it fits here.

File: src/sample-data/sample-data-generator.ts (hunk 1 above)
Comment (anshul0708) [inline src/sample-data/sample-data-generator.ts]: Yes the changes check if it is a two character language code first.

**Pull request merged**